core/domain/exp_domain.py

```diff
@@ -4295,6 +4346,12 @@ def _migrate_to_latest_yaml_version(
                 exploration_dict)
             exploration_schema_version = 29
 
+        if exploration_schema_version == 29:
+            exploration_dict['id'] = exp_id
```

Comment:
This seems wrong -- you shouldn't be modifying the exploration dict. Pass it in as a separate arg.

Reply:
Done